ocaml-eglot-type-enclosing.el

```diff
@@ -0,0 +1,157 @@
+  (interactive)
+  (if ocaml-eglot-type-enclosing-verbosity
+      (setq ocaml-eglot-type-enclosing-verbosity
+            (1+ ocaml-eglot-type-enclosing-verbosity))
+    (setq ocaml-eglot-type-enclosing-verbosity 1))
+  (ocaml-eglot-type-enclosing--with-fixed-offset))
+
+(defun ocaml-eglot-type-enclosing-grow ()
+  "Growing of the type enclosing."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (>= ocaml-eglot-type-enclosing-offset
+            (1- (length ocaml-eglot-type-enclosing-types)))
+        (setq ocaml-eglot-type-enclosing-offset 0)
+      (setq ocaml-eglot-type-enclosing-offset
+            (1+ ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing-shrink ()
+  "Shrinking of the type enclosing."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (<= ocaml-eglot-type-enclosing-offset 0)
+        (setq ocaml-eglot-type-enclosing-offset
+              (1- (length ocaml-eglot-type-enclosing-types)))
+      (setq ocaml-eglot-type-enclosing-offset
+            (1- ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing--type-buffer (type-expr)
+  "Create buffer with content TYPE-EXPR of the enclosing type buffer."
+  (let ((curr-dir default-directory)
+        (current-major-mode major-mode))
+    (with-current-buffer (get-buffer-create ocaml-eglot-type-buffer-name)
+      (funcall current-major-mode)
+      (read-only-mode 0)
+      (erase-buffer)
+      (insert type-expr)
+      (goto-char (point-min))
+      (read-only-mode 1)
+      (setq default-directory curr-dir))))
+
+(defun ocaml-eglot-type-enclosing--display (type-expr &optional current-enclosing)
+  "Display the type-enclosing for TYPE-EXPR in a dedicated buffer.
+If CURRENT-ENCLOSING is set, the range of the enclosing will be highlighted."
+  (ocaml-eglot-type-enclosing--type-buffer type-expr)
+  (if (ocaml-eglot-util--text-less-than type-expr 8)
+      (message "%s" (with-current-buffer ocaml-eglot-type-buffer-name
+                      (font-lock-fontify-region (point-min) (point-max))
+                      (buffer-string)))
+    (display-buffer ocaml-eglot-type-buffer-name))
+  (when current-enclosing
+    (let ((current-enclosing (aref ocaml-eglot-type-enclosing-types
+                                   ocaml-eglot-type-enclosing-offset)))
+      (ocaml-eglot-util--highlight-range current-enclosing
+                                         'ocaml-eglot-highlight-region-face))))
+
+(defun ocaml-eglot-type-enclosing--reset ()
+  "Reset local variables defined by the enclosing query."
+  (setq ocaml-eglot-type-enclosing-current-type nil)
+  (setq ocaml-eglot-type-enclosing-verbosity nil)
+  (setq ocaml-eglot-type-enclosing-types nil)
+  (setq ocaml-eglot-type-enclosing-offset 0))
+
+(defun ocaml-eglot-type-enclosing--call ()
+  "Prepare the type-enclosings computation request."
```

Comment:
I think the feature is named `type-enclosing` without an `s`.
But what it does is returning the type of enclosings with an `s`.

```suggestion
  "Prepare the type-enclosing computation request."
```

Reply:
Also, it's not only a preparation, but it displays a type to the user. This function and the other user-facing one should describe as clearly as possible what they are doing.

Reply:
See https://github.com/tarides/ocaml-eglot/pull/14/commits/05da9e520163310f309aca7e2fcbd631172e207c